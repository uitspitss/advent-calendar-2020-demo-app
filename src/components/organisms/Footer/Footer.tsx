import React, { FC } from 'react';
import tw, { styled } from 'twin.macro';

import { copyrightYears } from 'utils';

export type FooterProps = {
  children?: never;
  className?: string;
};

type Props = FooterProps;

const Component: FC<Props> = (props) => {
  const { className } = props;

  return (
    <footer className={className}>
      <div>
        <p>
          ©{copyrightYears(2020)} *WEBSITE TITLE* —
          <a
            href="https://twitter.com/uitspitss"
            rel="noopener noreferrer"
            target="_blank"
          >
            @uitspitss
          </a>
        </p>
      </div>
    </footer>
  );
};

const StyledComponent = styled(Component)`
  & > div {
    ${tw`flex items-center sm:flex-row flex-col py-2 px-2 lg:px-6`}

    & > p {
      ${tw`text-sm text-gray-500`}
      & > a {
        ${tw`text-gray-600 ml-1 no-underline`}
      }
    }
  }
`;

/**
 * フッター
 */
export const Footer: FC<FooterProps> = (props) => (
  <StyledComponent {...props} />
);
