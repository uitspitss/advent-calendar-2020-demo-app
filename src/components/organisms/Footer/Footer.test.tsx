import React from 'react';
import renderer from 'react-test-renderer';

import { Footer } from './Footer';

describe('Footer', () => {
  test('Snap Shot', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
