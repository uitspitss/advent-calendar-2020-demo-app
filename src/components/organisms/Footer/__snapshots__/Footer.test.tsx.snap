// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snap Shot 1`] = `
.emotion-0 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

@media (min-width:640px) {
  .emotion-0 > div {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:1024px) {
  .emotion-0 > div {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

.emotion-0 > div > p {
  font-size: 0.875rem;
  --text-opacity: 1;
  color: rgba(160,174,192,var(--text-opacity));
}

.emotion-0 > div > p > a {
  --text-opacity: 1;
  color: rgba(113,128,150,var(--text-opacity));
  margin-left: 0.25rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<footer
  className="emotion-0 emotion-1"
>
  <div>
    <p>
      ©
      2020
       
      *WEBSITE TITLE* —
      <a
        href="https://twitter.com/uitspitss"
        rel="noopener noreferrer"
        target="_blank"
      >
        @uitspitss
      </a>
    </p>
  </div>
</footer>
`;
